File: python/kode program/laptop.py
from elektronik import Elektronik   

class Laptop(Elektronik):         
    def __init__(self, id, nama, harga, kategori, merk, garansi, stok, warna,
                 prosesor, ram, storage, layar):
        super().__init__(id, nama, harga, kategori, merk, garansi, stok, warna)
        self.__prosesor = prosesor# Tipe prosesor (atribut private)
        self.__ram = ram # Kapasitas RAM  (atribut private)
        self.__storage = storage# Kapasitas storage (atribut private)
        self.__layar = layar# Ukuran layar dalam inci (atribut private)

    # Getter & Setter
    def get_prosesor(self): return self.__prosesor
    def set_prosesor(self, value): self.__prosesor = value

    def get_ram(self): return self.__ram
    def set_ram(self, value): self.__ram = value

    def get_storage(self): return self.__storage
    def set_storage(self, value): self.__storage = value

    def get_layar(self): return self.__layar
    def set_layar(self, value): self.__layar = value

File: java/kode program/Laptop.java
public class Laptop extends Elektronik {// Class Laptop turunan dari Elektronik
    private String prosesor;// Tipe prosesor
    private int ram;// Kapasitas RAM 
    private int storage;// Kapasitas storage
    private double layar;// Ukuran layar

    // Constructor dengan parameter
    public Laptop(int id,String nama,double harga,String kategori,
                  String merk,int garansi,int stok,String warna,
                  String prosesor,int ram,int storage,double layar) {
        super(id,nama,harga,kategori,merk,garansi,stok,warna); // Panggil constructor Elektronik
        this.prosesor=prosesor; this.ram=ram; this.storage=storage; this.layar=layar;
    }

    // Getter & Setter untuk atribut Laptop
    public String getProsesor() { return prosesor; }
    public void setProsesor(String prosesor) { this.prosesor=prosesor; }

    public int getRam() { return ram; }
    public void setRam(int ram) { this.ram=ram; }

    public int getStorage() { return storage; }
    public void setStorage(int storage) { this.storage=storage; }

    public double getLayar() { return layar; }
    public void setLayar(double layar) { this.layar=layar; }
}
